scanapi/tree/request_node.py

```diff
@@ -155,16 +155,21 @@ def run(self):
         url = self.full_url_path
         console.print(f"\n- Making request {method} {url}", highlight=False)
 
-        session = session_with_retry(self.retry)
-        response = session.request(
-            method,
-            url,
-            headers=self.headers,
-            params=self.params,
-            json=self.body,
-            allow_redirects=False,
+        kwargs = {
+            "headers": self.headers,
+            "params": self.params,
+            "json": self.body,
+            "allow_redirects": False,
             **self.options,
-        )
+        }
+
+        for k, v in self.headers.items():
```

Comment:
Maybe it would be nice to move this to a private method since this function is already huge. What do you think?